lms/templates/courseware/courseware.html

```diff
@@ -112,6 +112,47 @@
       </script>
   % endif
 
+  <script type="text/javascript">
+    /* Helper function isInViewport checks whether
+    the given element is in viewport vertically or not */
+    function isInViewport(el) {
+      const scroll = window.scrollY || window.pageYOffset
+      const elementTop = el.getBoundingClientRect().top + scroll
+      const viewport = {
+        top: scroll,
+        bottom: scroll + window.innerHeight,
+      }
+      const elementMid = elementTop + el.clientHeight/2
+      // Returns true if the middle of the element is in the viewport.
+      return elementMid >= viewport.top && elementMid <= viewport.bottom
+    }
+
+    /*Add a jQuery plugin to override the focus behavior.
+    When focused, the element will be the scrolled to bottom of the viewport.
+    */
+    (function ($) {
+      $.fn.extend({
+        focus: (function(orig) {
+          return function(delay, fn) {
+            orig.apply(this, arguments);
+            this.each(function(){
+              var elem = this;
+              // Scroll only when the element is not in the viewport and it contains the notification-btn.
+              if (!isInViewport(elem) && elem.classList.contains('notification-btn')) {
```

Comment:
It's a small thing, but if you reverse the clauses in this conditional then you won't do the viewport calculation for anything except the `notification-btn`.   If there are a lot of elements, it could reduce the work being done here.

```
if (elem.classList.contains('notification-btn') && !isInViewport(elem)) {
```

That'll short circuit if the element's class list doesn't contain `notification-btn`, avoiding the call to `isInViewport`.